File: 05_arrays/05_10_missing_number/missing_number.js
const missingNumber = (arr = []) => {
    let xor1 = 0;
    let xor2 = 0;

    for (let index = 0; index < arr.length; index++) {
        xor1 ^= arr[index];
        xor2 ^= index + 1;
    }

    return xor1 ^ xor2;
}

const nums = [9, 6, 4, 2, 3, 5, 7, 0, 1];

console.log("Elements:- ");
console.log(nums.join(' '));

const result = missingNumber(nums);
console.log("The missing number is:", result);

File: 05_arrays/05_10_missing_number/MissingNumber.java
public class MissingNumber {
    public static void main(String[] args) {
        int[] nums = { 9, 6, 4, 2, 3, 5, 7, 0, 1 };

        System.out.println("Elements:- ");
        for (int num : nums) {
            System.out.print(num + " ");
        }
        System.out.println();

        int result = missingNumber(nums);
        System.out.println("The missing number is: " + result);
    }

    private static int missingNumber(int arr[]) {
        int xor1 = 0;
        int xor2 = 0;

        for (int i = 0; i < arr.length; i++) {
            xor1 ^= arr[i];
            xor2 ^= (i + 1);
        }

        return xor1 ^ xor2;
    }
}
